core/src/org/pentaho/platform/engine/services/connection/datasource/dbcp/NonPooledDatasourceSystemListener.java

```diff
@@ -66,6 +66,8 @@ public boolean startup( final IPentahoSession session ) {
           if ( !databaseConnection.getAccessType().equals( DatabaseAccessType.JNDI ) ) {
             // if connection's port used by server there is no sense to get DataSource for this
             ds = isPortUsedByServer( databaseConnection ) ? null : setupDataSourceForConnection( databaseConnection );
+          } else {
+            ds = setupDataSourceForConnection( databaseConnection );
```

Comment:
Yea looks like if it's JNDI, setupDataSourceForConnection will catch an exception from PooledDatasourceHelper.convert and simply return null. Since we know it's JNDI ahead of time, I would rather simply "continue;" the loop and skip it entirely.